crates/provider-sdk/src/lib.rs

```diff
@@ -135,74 +132,21 @@ pub struct Context {
     pub tracing: HashMap<String, String>,
 }
 
-/// The super trait containing all necessary traits for a provider
-pub trait Provider: ProviderHandler + WrpcNats + WrpcDispatch + Send + Sync + 'static {}
-
 /// Configuration of a link that is passed to a provider
-pub trait LinkConfig: Send + Sync {
+pub struct LinkConfig<'a> {
     /// Given that the link was established with the source as this provider,
-    /// get the target ID which should be a component
-    fn get_target_id(&self) -> &LatticeTarget;
+    /// this is the target ID which should be a component
+    pub target_id: &'a str,
```

Comment:
We actually originally made LinkConfig a trait so that we could add or remove members to it without breaking the public API, do you think that wont be necessary?

Reply:
I actually think we're ok here as we will definitely need at least these fields. Anything else is additive, which we could get away with...probably. It would break the SDK, but not compatibility with wasmcloud hosts. Let's just keep it simple for now as if we have to add something to the trait it would break as well. tl;dr; I think I'm ok with this

Reply:
Agree

Reply:
I've marked the struct `non_exhaustive` to address the compatibility concerns here - for all other intents and purposes I struggle to see the difference in terms of compatibility compared to a trait here.

In terms of motivation for this change - like @thomastaylor312 outlined above, mostly simplicity and making developer experience smoother/easier.
Do note that this change actually fixed a bug https://github.com/wasmCloud/wasmCloud/blob/10ea04b4c36c31a16b20a3cfdffc56da5a39c948/crates/provider-keyvalue-vault/src/lib.rs#L368, which IMO is caused exactly by the trait abstraction

Apart from that the big reason for the change is maintainability - implementing this trait for opaque tuples is very error-prone and adding fields to those tuples would only make it worse